### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by toddsantoro** [jnweiger/led-name-badge-ls32#19]
I am getting a error `/usr/local/Cellar/python/3.7.4/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/MacOS/Python: can't open file './led-badge-11x44.py': [Errno 2] No such file or directory` when running the Hello World! Example on OSX 10.14.6. Any idea whats going on?

**Comment by jnweiger** [comment]
No idea. Sorry. 
Seems you have installed a python interpreter, seems you have the python script, so why the interpreter cannot find the  script, I cannot tell. Different folder?

**Comment by toddsantoro** [comment]
Do I need to be in the python folder to run the script?

Cheers,

Todd Santoro
Sr. Software Engineer
TSD

tel :: 305.699.3515
sms :: 305.699.3515
email :: todd@toddsantoro.com
web :: toddsantoro.com
dribbble: dribbble.com/toddsantoro
skype :: toddsantoro007
rss :: toddsantoro.com/feed
twitter :: @toddsantoro

**Comment by toddsantoro** [comment]
pip install pyhidapi
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already satisfied: pyhidapi in /Users/toddsantoro/Library/Python/2.7/lib/python/site-packages (0.0.2)

**Comment by toddsantoro** [comment]
pip install pyhidapi
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already satisfied: pyhidapi in /Users/xxxxxx/Library/Python/2.7/lib/python/site-packages (0.0.2)

**Comment by s-light** [comment]
try with `pip3 install pyhidapi`
this way you 'force' to install for python3.

**Comment by jnweiger** [comment]
yes, when you prefix the script name with ./ then that means "take it from
here" -- the current folder.

Todd Santoro <notifications@github.com> schrieb am Di., 10. Sep. 2019 15:03:

**Comment by toddsantoro** [comment]
I installed it with pip3 and this is the locations of the install
    /usr/local/bin/hidapi_enum.py
    /usr/local/lib/python3.7/site-packages/pyhidapi-0.0.2.dist-info/*
    /usr/local/lib/python3.7/site-packages/pyhidapi/*

**Comment by toddsantoro** [comment]
I am getting closer... I get this error now:
`ERROR: Need the pyhidapi or usb.core module.
Please try
  pip install pyhidapi
  brew install hidapi`

**Comment by toddsantoro** [comment]
I finally got it working. Thanks for the help and the code!!!

**Comment by Samplissimo** [comment]
hi todd, 
how did you get this working?
I have the same error: need the pyhidapi, but pyhidapi is installed -> 
djtrackmark@djtrackmark-Satellite-L300D:~/led-badge-44x11$ sudo -H pip install pyhidapi
Requirement already satisfied: pyhidapi in /usr/local/lib/python2.7/dist-packages

**Comment by toddsantoro** [comment]
you need to run the command from the directory of the led-name-badge folder

**Comment by jnweiger** [comment]
@Samplissimo it seems your pip installed the pyhidapi module for python2 -- do you run the script with python2 or python3?

**Comment by Samplissimo** [comment]
Hello all, 
Yes, i am running it from the Led-Badge folder.
Also i have the latest Python release and running the command with python 3.
Nevermind, i am new to Linux and i see this will be difficult - i managed to program the Badge via Windows interface, Neverless thanks for your hints.
Samplissimo

**Comment by jnweiger** [comment]
In this case you need pip3 install ...
Linux shows a lot of its internal logic to the end user. I've improved the
hints my script would print in the future. Thanks for the feedback!

Samplissimo <notifications@github.com> schrieb am Sa., 28. Dez. 2019 09:07:

**Comment by cnorrick** [comment]
> you need to run the command from the directory of the led-name-badge folder

I know this is a very old thread... but where is this folder? I've searched high and low. I used the 'pip3 install pyhidapi' and the 'brew install hidapi' with success (I think). But I don't see the led-badge-44x11.py script anywhere. And running the Hello World command yields the same "No such file directory" error. I've tried to run it from various locations. Including the Python folder.
